Makefile

```diff
@@ -33,29 +38,40 @@ gen_yaml_tests: $(PY_SPEC_ALL_TARGETS) $(YAML_TEST_TARGETS)
 install_test:
 	cd $(PY_SPEC_DIR); python3 -m venv venv; . venv/bin/activate; pip3 install -r requirements-testing.txt;
 
-test: $(PY_SPEC_ALL_TARGETS)
-	cd $(PY_SPEC_DIR); . venv/bin/activate; python -m pytest .
+test: test_phase0 test_phase1
+
+test_phase0: $(PY_SPEC_ALL_TARGETS)
+	cd $(PY_SPEC_DIR); . venv/bin/activate; python -m pytest tests/phase0
+
+test_phase1: $(PY_SPEC_ALL_TARGETS)
+	cd $(PY_SPEC_DIR); . venv/bin/activate; python -m pytest tests/phase1
 
 citest: $(PY_SPEC_ALL_TARGETS)
-	cd $(PY_SPEC_DIR); mkdir -p test-reports/eth2spec; . venv/bin/activate; python -m pytest --junitxml=test-reports/eth2spec/test_results.xml .
+	cd $(PY_SPEC_DIR); mkdir -p test-reports/eth2spec; . venv/bin/activate;	\
+	python -m pytest --junitxml=test-reports/eth2spec/test_results_phase0.xml tests/phase0; \
+	python -m pytest --junitxml=test-reports/eth2spec/test_results_phase0.xml tests/phase1
 
 install_lint:
-	cd $(PY_SPEC_DIR); python3 -m venv venv; . venv/bin/activate; pip3 install flake8==3.5.0
+	cd $(PY_SPEC_DIR); python3 -m venv venv; . venv/bin/activate; pip3 install flake8==3.7.0
 
 lint: $(PY_SPEC_ALL_TARGETS)
 	cd $(PY_SPEC_DIR); . venv/bin/activate; \
-	flake8 --max-line-length=120 ./eth2spec;
+	flake8 --max-line-length=120 --per-file-ignores='./tests/*:F821,F403,F401,F405' ./eth2spec ./tests;
```

Comment:
Linter messages `F821, F401, F403, F405` will always be raised as they are a function of how I designed the tests. They are only ignored for the `test_libs/pyspec/tests/*` files.

`F821` is fundamental to this redesign and there are too many occurrences to handle individually with `# noqa`.
The linter doesn't raise any issues around the use of `spec.foo()` and `helpers.bar()` as, from its perspective, `spec` and `helpers` are undefined.

`F401, F403, F405` These could be ignored individually at the time of import, but I this is cleaner.


Are there any objections to doing it this way? Sidenote: using `--per-file-ignores='...'` is a feature that requires flake8 > v3.7.0. is there any reason why I can't use this version?

Reply:
Personally I'm ok with this. I think it's better than applying some other machinery to effectively circumvent the linter.